pkg/metrics/prometheus.go

```diff
@@ -0,0 +1,152 @@
+package metrics
+
+import (
+	"github.com/grafana/grafana/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+	"strings"
+)
+
+const (
+	namespace = "grafana"
+	subsystem = "backend"
+)
+
+type PrometheusPublisher struct {
+	CounterCollectors map[string]*prometheus.CounterVec
+	GaugeCollectors   map[string]*prometheus.GaugeVec
+}
+
+func CreatePrometheusPublisher() (*PrometheusPublisher, error) {
+	go func() {
+		http.Handle("/metrics", prometheus.Handler())
+		err := http.ListenAndServe(":9200", nil)
```

Comment:
Oh, and forgot to mention the metrics api endpoint return counters that never reset (as that would not make sense, as the consumer needs to control the sample period). It's only the metric publishers that sends data at a specified interval that get's counters that are deltas since last period. 

So making another implementation of this:
https://github.com/grafana/grafana/blob/master/pkg/api/metrics.go#L60

that returns prometheus format sounds like a good idea.